corporate/lib/stripe.py

```diff
@@ -256,6 +256,30 @@ def make_end_of_cycle_updates_if_needed(plan: CustomerPlan,
                 plan=plan, is_renewal=True, event_time=next_billing_cycle,
                 licenses=last_ledger_entry.licenses_at_next_renewal,
                 licenses_at_next_renewal=last_ledger_entry.licenses_at_next_renewal)
+
```

Comment:
We are inside `if next_billing_cycle <= event_time:` block here. I don't quite understand what that if is doing, and if it's okay for us here in the switch plan case?

Reply:
`next_billing_cycle` would the time in which the plans gets renewed. For monthly plan this happens every month and for annual plan this happens every 12 months.

In our case, we have to switch the monthly plan to annual at `next_billing_cycle`. Thats why we do it here.

Does that makes sense?

Reply:
Yup, that makes sense!